### PR TITLE
Reduce memory allocations, remove unused constants, move to event id,…

### PR DESCRIPTION
**Pull request by Eraden**
* Reduce memory allocations
* Remove unused constants
* Move to event id (https://github.com/red-life-project/red-life/issues/275)
* Remove redundant enum values
* Fix const name issue (https://github.com/red-life-project/red-life/issues/276)

### CODE REVIEW

File: game/src/backend/screen.rs [hunk 1]
```diff
@@ -54,7 +55,7 @@ pub struct ScreenStack {
 #[derive(Debug, PartialEq, Clone)]
 pub struct Popup {
     color: Color,
-    text: String,
+    text: Cow<'static, str>,
```
Comment (Nereuxofficial) [inline game/src/backend/screen.rs:58]: Good catch with the Copy-on-write Types. Didn't think of it at the time

File: game/src/game_core/event.rs [hunk 2]
```diff
@@ -177,53 +166,47 @@ impl Event {
...
     /// * `context` - The game context which is used to access the current tick
-    pub fn update_events(ctx: &Context, gamestate: &mut GameState) -> RLResult {
-        let lng = gamestate.lng;
+    pub fn update_events(ctx: &Context, game_state: &mut GameState) -> RLResult {
         if ctx.time.ticks() % 20 == 0 {
-            gamestate.events.iter_mut().for_each(|event| {
+            game_state.events.iter_mut().for_each(|event| {
                 event.duration = event.duration.saturating_sub(20);
-                if event.name == "Sandsturm" {}
+                if event.event_id == EventId::Sandstorm {}
```
Comment (Nereuxofficial) [inline game/src/game_core/event.rs:193]: This line seems to be an artifact that's no longer needed

**Review by Nereuxofficial — APPROVED** [review]
Very nice cleanups and improvements, especially the Cow parts. LGTM!

File: game/src/game_core/event.rs (hunk 2 above)
Comment (Eraden) [reply game/src/game_core/event.rs:193]: I was convinced this is something TO BE IMPLEMENTED

Comment (Nereuxofficial) [reply game/src/game_core/event.rs:193]: Maybe it was but it was never implemented and i do not recall what was to be implemented and there is no comment left here.
This was a project for our university and sadly we do not have the time to improve the game much anymore. I tried porting it to webassembly with good-web-game which worked reasonably well after the project was finished and made a MacOS release but im afraid apart from your major improvements that's the only development the game has seen as the team all has stressful jobs

**Pull request merged**